skbio/sequence/_sequence.py

```diff
@@ -1230,21 +1303,55 @@ def copy(self, deep=False):
         2   []        8
         3   []        5
 
+        >>> interval.metadata['gene'] = ['BCC7']
+        >>> seq.interval_metadata   # doctest: +ELLIPSIS
+        1 interval feature
+        ------------------
+        Interval(interval_metadata=<...>, bounds=[(0, 2)], \
+fuzzy=[(False, False)], metadata={'gene': ['BCC7']})
+        >>> interval.bounds = [(1, 3)]
+        >>> seq.interval_metadata  # doctest: +ELLIPSIS
+        1 interval feature
+        ------------------
+        Interval(interval_metadata=<...>, bounds=[(1, 3)], \
+fuzzy=[(False, False)], metadata={'gene': ['BCC7']})
+        >>> seq_copy.interval_metadata   # doctest: +ELLIPSIS
+        1 interval feature
+        ------------------
+        Interval(interval_metadata=<...>, bounds=[(0, 2)], \
+fuzzy=[(False, False)], metadata={'gene': 'p53'})
+
+        >>> interval2 = seq_copy.interval_metadata.add(
+        ...     [(2, 3)], metadata={'gene': 'brca1'})
+        >>> seq_copy.interval_metadata   # doctest: +ELLIPSIS
+        2 interval features
+        -------------------
+        Interval(interval_metadata=<...>, bounds=[(0, 2)], \
+fuzzy=[(False, False)], metadata={'gene': 'p53'})
+        Interval(interval_metadata=<...>, bounds=[(2, 3)], \
+fuzzy=[(False, False)], metadata={'gene': 'brca1'})
+        >>> seq.interval_metadata  # doctest: +ELLIPSIS
+        1 interval feature
+        ------------------
+        Interval(interval_metadata=<...>, bounds=[(1, 3)], \
+fuzzy=[(False, False)], metadata={'gene': ['BCC7']})
+
         Since only a *shallow* copy was made, updates to mutable objects stored
         as metadata affect the original sequence's metadata:
 
+        >>> seq_copy = seq.copy()
         >>> seq_copy.metadata['authors'].append('Bob')
         >>> pprint(seq_copy.metadata)
-        {'authors': ['Alice', 'Bob'], 'id': 'new-id'}
+        {'authors': ['Alice', 'Bob'], 'id': 'seq-id'}
```

Comment:
because it is a new copy, not the copy that was modified. pls see the change in the doctest.